plugins/merge_RGB.py

```diff
@@ -0,0 +1,314 @@
+        self._dlg.addStream(stream_ch)
+        self._setupStreambar()
+
+        self._cropBottom()
+        self._connectShift(stream_ch, 0, shiftX)
+        self._connectShift(stream_ch, 1, shiftY)
+
+        return stream_ch
+
+    def _ensureRGB(self, data, tint):
+        """
+        Ensures that the image is grayscale. If the image is a grayscale RGB, convert it
+        to an 8bit grayscale image of 2 dimensions and assign the corresponding tint to it.
+        Update the metadata of the image.
+        data (DataArray or DataArrayShadow): The input image
+        return (DataArray): The result image which the assigned tint
+        raises: ValueError if the image is RGB with different color channels
+        """
+        if len(data.shape) > 3:
+            raise ValueError("Image format not supported")
+        elif len(data.shape) == 3:
+            if isinstance(data, model.DataArrayShadow):
+                data = data.getData()
+            data = img.ensureYXC(data)
+            if (numpy.all(data[:, :, 0] == data[:, :, 1]) and
+                numpy.all(data[:, :, 0] == data[:, :, 2])):
+                data = data[:, :, 0]
+                data.metadata[model.MD_DIMS] = "YX"
+            else:
+                raise ValueError("Coloured RGB image not supported")
+        data.metadata[model.MD_USER_TINT] = tint
+
+        return data
+
+    def _connectShift(self, stream, index, vashift):
+        """Create listeners with information of the stream and the dimension.
+        Hold a reference to the listeners to prevent automatic subscription"""
+        va_on_shift = functools.partial(self._onShift, stream, index)
+        self._subscribers.append(va_on_shift)
+        vashift.subscribe(va_on_shift)
+
+    def _removeStream(self, st):
+        """Remove the given stream since another one is loaded from the user for display"""
+        sconts = self._dlg.streambar_controller.stream_controllers
+        for sc in sconts:
+            if sc.stream is st:
+                sc.stream_panel.on_remove_btn(st)
+
+    @call_in_wx_main
+    def _setupStreambar(self):
+        """Force stream panel to static mode. Needed for preventing user to play or
+        remove streams from the stream panel"""
+        sconts = self._dlg.streambar_controller.stream_controllers
+        for sctrl in sconts:
+            sctrl.stream_panel.to_static_mode()
+
+    def _onShift(self, stream, i, value):
+        """
+        Update the stream after shifting it by the given value.
+        Args:
+            stream(StaticFluoStream): stream to be shifted
+            i(int): index to show at which dimension the stream is to be shifted
+            value(ContinuousVA): shift values in meters
+        """
+        logging.debug("New shift = %f on stream %s", value, stream.name.value)
+        poscor = stream.raw[0].metadata.get(model.MD_POS_COR, (0, 0))
+        px_size = stream.raw[0].metadata.get(model.MD_PIXEL_SIZE)
```

Comment:
You have to provide some default value for the PIXEL_SIZE. Otherwise, you might get `None`, which will mess up the rest.
I'd suggest to extend `_ensureRGB()` so that it detects there is no such metadata, and in such case add 1e-9x1e-9 (as before). Then, here you can just write `px_size = stream.raw[0].metadata[model.MD_PIXEL_SIZE]`.